sieve/message.c

```diff
@@ -172,7 +172,8 @@ int do_fileinto(action_list_t *a, const char *mbox, const char *specialuse,
             return SIEVE_RUN_ERROR;
         }
 
-        if (a->a == ACTION_FILEINTO && !strcmp(a->u.fil.mailbox, mbox)) {
+        if ((a->a == ACTION_FILEINTO && !strcmp(a->u.fil.mailbox, mbox)) ||
+            ((a->a == ACTION_KEEP && !strcmp("INBOX", mbox)))) {
```

Comment:
As Neil commented - shouldn't we be implicit keeping into the potentially plus-addressed destination mailbox rather than "INBOX" specifically?